src/__tests__/DdSdkReactNative.test.tsx

```diff
@@ -53,14 +53,14 @@ jest.mock('../rum/instrumentation/DdRumErrorTracking', () => {
 
 beforeEach(async () => {
     DdSdkReactNative['wasInitialized'] = false;
-    NativeModules.DdSdk.initialize.mockReset()
-    NativeModules.DdSdk.setAttributes.mockReset()
-    NativeModules.DdSdk.setUser.mockReset()
-    NativeModules.DdSdk.setTrackingConsent.mockReset()
-
-    DdRumUserInteractionTracking.startTracking.mockReset()
-    DdRumResourceTracking.startTracking.mockReset()
-    DdRumErrorTracking.startTracking.mockReset()
+    NativeModules.DdSdk.initialize.mockClear()
```

Comment:
`mockReset()` -> `mockClear` change is needed, because according to https://jestjs.io/docs/mock-function-api#mockfnmockreset the former also removes return values (which were set with calls like `jest.fn().mockImplementation(() => { })` before) making mocks useless.

Reply:
But what if you have mocked a return value in one of your tests which will basically override the default one -> `null` or `()=> { }`. This can cause side effects on your next tests case right ?

Reply:
it seems like no. I guess this is because second argument of `jest.mock` is actually the factory, which is called before each test.